source/bson-corpus/bson-corpus.rst

```diff
@@ -338,6 +338,27 @@ Implementations MAY ignore or modify them to match legacy treatment of
 deprecated types.  The ``converted_bson`` and ``converted_extjson`` fields MAY
 be used to test conversion to a standard type or MAY be ignored.
 
+Prose Tests
+===========
+
+The following tests have not yet been automated, but MUST still be tested.
+
+1. Prohibit null bytes in null-terminated strings when encoding BSON
+--------------------------------------------------------------------
+
+The BSON spec uses null-terminated strings to represent document field names and
+regex components (i.e. pattern and flags/options). Drivers MUST assert that null
+bytes are prohibited in the following contexts when encoding BSON.
+
+* Field name within a root document
+* Field name within a sub-document
+* Pattern for a regular expression
+* Flags/options for a regular expression
```

Comment:
We already have `decodeErrors` tests for regex types. The format of those tests is a hex string of raw BSON data that produces an exception when converting the BSON into a language type.

On the off chance you meant to ask about `parseErrors`: Apart from Decimal128, `parseErrors` tests typically expect errors when parsing extended JSON. It's unclear if the tests expect errors when converting extended JSON to BSON or directly to a native language type. In libmongoc, parsing extended JSON produces a `bson_t` (basically raw BSON). Since PHP uses that directly (i.e. JSON is converted to a BSON byte string), it's JSON-parsing functions do not directly produce PHP types.

To recap, `decodeErrors` already serves a different purpose. And I don't think `parseErrors` is particularly well suited to this kind of test, despite how it is presently used for Decimal128. One could argue there that the tests are "parsing" a string into a Decimal128 BSON type (or the language type to represent it), but I don't think that definition extends to other types. In the case of Regex, we're just soliciting strings and validating them. I think it's a stretch to call that "parsing".

I piggybacked the existing `parseErrors` tests in `top.json` for regex since it was already being used to test extended JSON errors for other BSON types (including some other regex cases).

Technically, we _could_ introduce new `parseErrors` syntax for regex to test this, but I think it's a poor name. And we'd still need prose tests to cover the document and sub-document cases (I can't imagine what a generalized spec test format would look like to accomplish those).

Reply:
I added a separate commit to this PR to clarify type-specific rules for `parseErrors`.